samcli/lib/telemetry/event.py

```diff
@@ -155,16 +158,25 @@ def track_event(event_name: str, event_value: str):
             should_send: bool = False
             with EventTracker._event_lock:
                 EventTracker._events.append(Event(event_name, event_value))
+                # Get the session ID (needed for multithreading sending)
+                if not EventTracker._session_id:
+                    try:
+                        ctx = Context.get_current_context()
+                        if ctx:
+                            EventTracker._session_id = ctx.session_id
+                            LOG.debug("EventTracker: Obtained session ID '%s'", EventTracker._session_id)
```

Comment:
Thanks for updating. But I believe it's more useful to log in the failure path (i.e. in the except block) than in the happy path.